src/celeritas/track/detail/TrackSortUtils.cu

```diff
@@ -29,6 +33,24 @@ void fill_track_slots<MemSpace::device>(Span<TrackSlotId::size_type> track_slots
         0);
     CELER_DEVICE_CHECK_ERROR();
 }
+
+/*!
+ * Shuffle track slots
+ */
+template<>
+void shuffle_track_slots<MemSpace::device>(
+    Span<TrackSlotId::size_type> track_slots)
+{
+    using result_type = thrust::default_random_engine::result_type;
+    thrust::default_random_engine g{
+        static_cast<result_type>(track_slots.size())};
+    thrust::shuffle(
+        thrust::device,
+        thrust::device_pointer_cast(track_slots.data()),
+        thrust::device_pointer_cast(track_slots.data() + track_slots.size()),
+        g);
```

Comment:
Using a different RNG here means we lose host/device reproducibility for the same number of tracks (probably not the worst, since we'll likely only use this for testing purposes). I wonder if, since we're initializing and shuffling the tracks only once per run (effectively), we should just construct (linear, then possibly sort) on the host and then (if necessary) shuffle them before sending to the device...

Reply:
Yes I suppose we could fill and shuffle the array on the host before sending it to the device. Is there any advantages compared the current implementation supposing we use the same RNG for `thrust::shuffle`? It seems pretty much equivalent to me.

Reply:
Not really: just simpler and less code. Since we're not realistically going to shuffle on host I'm cool with leaving as is.